src/cpp/database/database_queue.cpp

```diff
@@ -0,0 +1,645 @@
+                //TODO: maybe some logging would be nice
+            }
+        }
+        break;
+        case StatisticsEventKind::SAMPLE_DATAS:
+        {
+            SampleDatasCountSample sample;
+            EntityId domain;
+            EntityId entity;
+            queue_item_type item = front();
+            sample.src_ts = item.first;
+            try
+            {
+                process_sample_type(domain, entity, EntityKind::DATAWRITER, sample,
+                        item.second->sample_identity_count());
+                database_->insert(domain, entity, sample);
+            }
+            catch (const eprosima::statistics_backend::Exception& e)
+            {
+                //TODO: maybe some logging would be nice
+            }
+        }
+        break;
+        case StatisticsEventKind::PHYSICAL_DATA:
+        {
+            StatisticsPhysicalData item = front().second->physical_data();
+
+            // Take the ID of the Participant from its GUID
+            std::string participant_guid = deserialize_guid(item.participant_guid());
+            auto participants = database_->get_entities_by_guid(EntityKind::PARTICIPANT, participant_guid);
+            if (participants.empty())
+            {
+                std::stringstream msg;
+                msg << "No participant with GUID " << participant_guid << " exists";
+                throw BadParameter(msg.str());
+            }
+            EntityId participant_id = participants.front().second;
+
+            // Parse the process name and PID
+            size_t separator_pos = item.process().find_last_of(':');
+            if (separator_pos == std::string::npos)
+            {
+                std::stringstream msg;
+                msg << "Process name " << item.process() << " does not follow the [command]:[PID] pattern";
+                throw Error(msg.str());
+            }
+            std::string process_name = item.process().substr(0, separator_pos);
+            std::string process_pid = item.process().substr(separator_pos + 1);
+
+            // Check the existence of the process
+            EntityId process_id;
+            std::shared_ptr<Process> process;
+            auto processes = database_->get_entities_by_name(EntityKind::PROCESS, process_name);
+            if (processes.empty())
+            {
+                // Check the existence of the user
+                std::shared_ptr<User> user;
+                auto users = database_->get_entities_by_name(EntityKind::USER, item.user());
+                if (users.empty())
+                {
+                    // Check the existence of the host
+                    std::shared_ptr<Host> host;
+                    auto hosts = database_->get_entities_by_name(EntityKind::HOST, item.host());
+                    if (hosts.empty())
+                    {
+                        host.reset(new Host(item.host()));
+                        database_->insert(std::static_pointer_cast<Entity>(host));
```

Comment:
Doing this here means that when adding a host using the discovery callbacks, we must first check if we have it already. Currently, there is no mechanism for that, because I don't think host name is unique enough, unless we make it so from Fast DDS somehow (maybe we should!).